dissect/target/filesystems/config.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import annotations
+
+import io
+import textwrap
+from typing import Any, BinaryIO, Iterator, Optional, Union
+
+from dissect.target import Target
+from dissect.target.exceptions import ConfigurationParsingError
+from dissect.target.filesystem import Filesystem, FilesystemEntry, VirtualFilesystem
+from dissect.target.helpers import fsutil
+from dissect.target.helpers.configparser import (
+    ConfigurationParser,
+    ParserOptions,
+    parse_config,
+)
+
+
+def create_entry(
+    fs: Filesystem,
+    path: str,
+    entry: FilesystemEntry,
+    hint: str,
+    collapse: str,
+    seperator: tuple[str],
+    comment_prefixes: tuple[str],
+) -> Union[FilesystemEntry, ConfigurationParser]:
+    if entry.is_file():
+        contents = ParserOptions(collapse, seperator, comment_prefixes)
+        parser_items = parse_config(entry, hint, contents)
+        return ConfigurationEntry(fs, path, entry, parser_items=parser_items)
+    return entry
+
+
+class ConfigurationFilesystem(VirtualFilesystem):
+    __fstype__: str = "META:configuration"
+
+    def __init__(self, target: Target, path: str = "/etc", **kwargs):
```

Comment:
```suggestion
    def __init__(self, target: Target, path: str, **kwargs):
```

I wouldn't set a default here. Also, the config and etc plugins pass in explicit paths anyway.